Fixed problems caused by a rotated simulator

ohammersmith opened the pull request:
- undefined constant
- iPhone 6/6plus folders would collide

I didn't see any existing tests and I'm not entirely sure how I would go about covering this with tests. Any suggestions welcome.

Thanks!

squidpunch commented on lib/juxtapose/strategy/mac_bacon_strategy.rb:
```diff
@@ -16,9 +16,9 @@ def current_spec_description
-      name << '5' if UIScreen.mainScreen.bounds.size.height == 568.0
-      name << '6' if UIScreen.mainScreen.bounds.size.height == 667.0
-      name << '6-plus' if UIScreen.mainScreen.bounds.size.height == 736.0
+      name << '5' if UIScreen.mainScreen.bounds.size.height == 568.0 || UIScreen.mainScreen.bounds.size.height == 320.0
```
we'd only want the 5 added if its 320 x 568, so we'd need to check both directions if the screen is rotated, otherwise we're going to collide with a 4 screenshot, right?

ohammersmith replied:
Yes you're right. 

I knew I was missing something. I hadn't included the 4 sizes in my test scripts. I'll update the fix. 

I wonder if it makes sense to incorporate size classes somehow. 

> On Jun 4, 2015, at 1:30 PM, David Larrabee <notifications@github.com> wrote:
> 
> In lib/juxtapose/strategy/mac_bacon_strategy.rb:
> 
> > @@ -16,9 +16,9 @@ def current_spec_description
> >      def device_name
> >        name = [UIDevice.currentDevice.model.gsub(/\s+Simulator/, '').downcase]
> >        name << 'retina' if UIScreen.mainScreen.scale > 1
> > -      name << '5' if UIScreen.mainScreen.bounds.size.height == 568.0
> > -      name << '6' if UIScreen.mainScreen.bounds.size.height == 667.0
> > -      name << '6-plus' if UIScreen.mainScreen.bounds.size.height == 736.0
> > +      name << '5' if UIScreen.mainScreen.bounds.size.height == 568.0 || UIScreen.mainScreen.bounds.size.height == 320.0
> we'd only want the 5 added if its 320 x 568, so we'd need to check both directions if the screen is rotated, otherwise we're going to collide with a 4 screenshot, right?
> 
> —
> Reply to this email directly or view it on GitHub.
>

ohammersmith commented:
There, should be fixed. I added iPhone 4 to my project and it worked alright.

Also I moved the extra long lines into private predicates. There may be existing predicates already in scope, I didn't go digging for it... it might be MotionKit and I didn't want to add dependencies.

squidpunch commented:
looks good to me, I'll merge, but I'll need to ping @thegreatape about releasing to rubygems!

squidpunch commented:
and yeah the only way you can test it is going out and mocking sizes, and check the file paths...stuff like that, or maybe a travis script that runs a subset of tests for each device type - but that feels slightly overkill to me.

If you really want to add a test, you could take the existing screenshot test and rotate it and make sure that one passes 

something like this

https://github.com/terriblelabs/motion-juxtapose/blob/master/spec/controllers/test_controller_spec.rb#L17-L19

but doing a rotation and then doing the screenshots

squidpunch commented:
just released 0.3.1 with your fix!

thanks @ohammersmith

ohammersmith commented:
Awesome, thanks!

I almost shouldn't have bothered pointing my `Gemfile` at my fork. 😊

jamonholmgren commented:
@squidpunch You're an owner of this? Didn't realize that.

squidpunch commented:
@jamonholmgren I wasn't until about 5 minutes before I send the release

I know @thegreatape and he added me as an owner so I could start sending out releases :smile:
